File: 프로그래머스/lv0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.py
def solution(numbers):
    #numbers.sort()
    answer = -10000*10000
    for i in range(len(numbers)-1):
        for j in range(i+1,len(numbers)):
            if numbers[i]*numbers[j]  >= answer:
                answer = numbers[i]*numbers[j]
            
    return answer

File: 프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.java
class Solution {
    public int solution(int[] numbers) {
        int answer =  Integer.MIN_VALUE; 
        for (int i = 0; i < numbers.length - 1; i++) {
            for (int j = i + 1; j < numbers.length; j++) {
                answer = findMax(answer, numbers[i], numbers[j]); // answer 업데이트
            }
        }
        return answer;
    }
    
    public int findMax(int currentMax, int num1, int num2) {
        int mul = num1 * num2;
        if (mul > currentMax) {
            return mul; // 최대값 반환
        }
        return currentMax; // 최대값 유지
    }
}
